hera_cal/nucal.py

```diff
@@ -724,32 +737,49 @@ def _linear_fit(XTX, Xy, solver='lu_solve', tol=1e-15):
     # If XTX is a jax array, use the jax array indexing syntax
     if (solver == "lu_solve" or solver == "solve") and isinstance(XTX, jnp.ndarray):
         XTX = XTX.at[np.diag_indices_from(XTX)].add(tol)
-    elif (solver == "lu_solve" or solver == "solve"):
+    elif solver == "lu_solve" or solver == "solve":
         XTX[np.diag_indices_from(XTX)] += tol
 
-    if solver == 'lu_solve':
+    if solver == "lu_solve":
         # Factor XTX using scipy.linalg.lu_factor
-        L = linalg.lu_factor(XTX)
+        if "LU" in cached_input:
+            L = cached_input.get('LU')
```

Comment:
Looks like you missed testing using the cache for lu_solve... should be add a test where you just run the smoother again.

Reply:
I ended up testing this in test_nucal